frontend/demo/src/actions/conversationsFilter/index.ts

```diff
@@ -0,0 +1,107 @@
+  resolve => (filter: ConversationFilter) => resolve({filter})
+);
+
+export const setSearch = (currentFilter: ConversationFilter, displayName: string) => {
+  return setFilter({
+    ...currentFilter,
+    displayName,
+  });
+};
+
+export const setFilter = (filter: ConversationFilter) => {
+  return (dispatch: Dispatch<any>, state: () => StateModel) => {
+    executeFilter(filter, dispatch, state);
+  };
+};
+
+export const resetFilter = () => {
+  return function(dispatch: Dispatch<any>, state: () => StateModel) {
+    dispatch(resetFilteredConversationAction());
+    const currentFilter = state().data.conversations.filtered.currentFilter;
+    const newFilter: ConversationFilter = {displayName: currentFilter.displayName};
+    executeFilter(newFilter, dispatch, state);
+  };
+};
+
+const executeFilter = (filter: ConversationFilter, dispatch: Dispatch<any>, state: () => StateModel) => {
+  dispatch(updateFilteredConversationsAction(filter));
+  refetchConversations(dispatch, state);
+};
+
+const refetchConversations = (dispatch: Dispatch<any>, state: () => StateModel, cursor?: string) => {
+  dispatch(loadingConversationsAction());
+  const filter = state().data.conversations.filtered.currentFilter;
+  return HttpClientInstance.listConversations({
+    page_size: 10,
+    cursor,
+    filters: filterToLuceneSyntax(filter),
+  })
+    .then((response: {data: Conversation[]; metadata: ResponseMetadataPayload}) => {
+      dispatch(mergeFilteredConversationsAction(response.data, filter, response.metadata));
+      return Promise.resolve(true);
+    })
+    .catch((error: Error) => {
+      return Promise.reject(error);
+    });
+};
+
+const filterToLuceneSyntax = (filter: ConversationFilter): string | null => {
+  let filterQuery = '';
+  if (filter.unreadOnly) {
+    filterQuery = 'unread_count:[1 TO *]';
+  } else if (filter.readOnly) {
+    filterQuery = 'unread_count:0';
+  }
+  if (filter.displayName) {
+    if (filterQuery === '') {
+      filterQuery = 'display_name:*' + filter.displayName + '*';
+    } else {
+      filterQuery += ' AND display_name:*' + filter.displayName + '*';
+    }
+  }
+  if (filter.byTags && filter.byTags.length > 0) {
+    if (filterQuery === '') {
+      filterQuery = 'tagId:(' + filter.byTags.join(' ') + ')';
+    } else {
+      filterQuery += ' AND tagId:(' + filter.byTags.join(' ') + ')';
+    }
+  }
+  if (filter.byChannels && filter.byChannels.length > 0) {
+    if (filterQuery === '') {
+      filterQuery = 'channelId:(' + filter.byChannels.join(' ') + ')';
+    } else {
+      filterQuery += ' AND channelId:(' + filter.byChannels.join(' ') + ')';
+    }
+  }
+  if (filter.bySources && filter.bySources.length > 0) {
+    if (filterQuery === '') {
+      filterQuery = 'source:(' + filter.bySources.join(' ') + ')';
+    } else {
+      filterQuery += ' AND source:(' + filter.bySources.join(' ') + ')';
+    }
+  }
+
+  console.log(filterQuery);
+
+  return filterQuery === '' ? null : filterQuery;
```

Comment:
if we introduce a `let filters = []`  we can do `filters.join("AND ")` to convert it back into a string right at the end. It should simplify the code quite a bit as we'd be able to save half of the ifs

Also console.log 💣